modules/ROOT/pages/traversal-framework/traversal_framework_java_api.adoc

```diff
@@ -0,0 +1,353 @@
+:description: The Neo4j Traversal Framework Java API.
+
+:org-neo4j-graphdb-Direction-both: {neo4j-javadocs-base-uri}/org/neo4j/graphdb/Direction.html#BOTH
+:org-neo4j-graphdb-traversal-description: {neo4j-javadocs-base-uri}/org/neo4j/graphdb/Transaction.html#traversalDescription()
+
+[[traversal-java-api]]
+= Traversal Framework Java API
+
+[[traversal-java-api-traversaldescription]]
+== link:{org-neo4j-graphdb-traversal-description}[TraversalDescription^]
+
+The `TraversalDescription` is the main interface used for defining and initializing traversals.
+It is not meant to be implemented by users of the Traversal Framework, but rather is provided as a way for the user to describe traversals.
+`TraversalDescription` instances are immutable and its methods return a new `TraversalDescription` that is modified compared to the object the method was invoked on with the arguments of the method.
+
+
+=== Relationships
+
+Adds a relationship type to the list of relationship types to traverse.
+By default, this list is empty meaning that it will traverse _all relationships_, regardless of type.
+If one or more relationships are present _only the added_ types will be traversed.
+There are two methods, one `including direction` and another `excluding direction`, where the latter traverses relationships in link:{org-neo4j-graphdb-Direction-both}[both directions^].
+
+
+[[traversal-java-api-evaluator]]
+== Evaluator
```

Comment:
```suggestion
=== Evaluator
```

Reply:
I assumed this was a sub-section of `TraversalDescription`, as it looked like a list of its methods. If not ignore. 

If it is then it would make sense to make the sub-section headings at least match the methods. So BranchSelector should probably be `order`, `PathExpander` => `expand` etc. And it might make it clearer to separate out `traverse` from the definition methods.

Reply:
Essentially everything in this file is to do with the TraversalDescription, based on some other comments we have rearranged some areas, but think the sub headings in general are fine as they are :)